### PR TITLE
hsMessageBox Cleanups

### PR DESCRIPTION
**Pull request by dpogue**
* Moves hsMessageBox implementation out of CoreLib and into PubUtilLib
* Splits the Win32-specific implementation into its own file (matching how macOS is handled)
* Adds overloads for `hsMessageBox` that take ST::string
* Silences (on gcc & clang) deprecation warnings from OpenSSL about RC4

I'm unsure on whether it makes sense to change the hsMessageBox stuff to always convert to ST::string and then have a single implementation, or if it's more performant having it split up as it currently is...

### CODE REVIEW

File: Sources/Plasma/CoreLib/HeadSpin_Win.cpp [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+    case IDIGNORE:      return hsMBoxIgnore;
+    case IDYES:         return hsMBoxYes;
+    case IDNO:          return hsMBoxNo;
+    default:            return hsMBoxCancel;
+    }
+
+    return hsMBoxCancel;
+}
+
+int hsMessageBoxWithOwner(hsWindowHndl owner, const char* message, const char* caption, int kind, int icon)
```
Comment (zrax) [inline Sources/Plasma/CoreLib/HeadSpin_Win.cpp:119]: Why not just pass these variants onto the `ST::string` variant, and then eventually deprecate/remove the non-`ST::string` overloads?  Otherwise, this is a lot of duplication...

Comment (dpogue) [reply Sources/Plasma/CoreLib/HeadSpin_Win.cpp:119]: That's what I was thinking, and performance of string conversions probably doesn't matter much in the case of message boxes

File: Sources/Plasma/CoreLib/HeadSpin_Mac.mm [hunk 2]
```diff
@@ -101,39 +104,19 @@ int hsMessageBoxWithOwner(hsWindowHndl owner, NSString* message, NSString* capti
...
-                                     [[[NSString alloc] initWithBytesNoCopy:(void *)caption
-                                                                     length:wcslen(caption)*sizeof(*caption)
-                                                                   encoding:NSUTF32LittleEndianStringEncoding freeWhenDone:NO
-                                      ] autorelease],
-                                     [[[NSString alloc] initWithBytesNoCopy:(void *)message
-                                                                     length:wcslen(caption)*sizeof(*caption)
-                                                                   encoding:NSUTF32LittleEndianStringEncoding freeWhenDone:NO
-                                      ] autorelease],
+                                     [NSString initWithUTF8String:message.c_str()],
+                                     [NSString initWithUTF8String:caption.c_str()],
```
Comment (dgelessus) [inline Sources/Plasma/CoreLib/HeadSpin_Mac.mm:119]:
```suggestion
                                     [NSString stringWithSTString:message],
                                     [NSString stringWithSTString:caption],
```

**Review by dgelessus — CHANGES_REQUESTED** [review]

**Comment by dpogue** [comment]
> I think it would make sense to pull `hsMessageBox` out into its own header. Possibly even into another library, somewhere under PubUtilLib.

I like this idea

File: Sources/Plasma/CoreLib/HeadSpin_Mac.mm (hunk 2 above)
Comment (dpogue) [reply Sources/Plasma/CoreLib/HeadSpin_Mac.mm:119]: We can't use that here (reliably) because NSString+STString category is in plClient, and this might (hypothetically) get called from somewhere that's not part of plClient.

Sure would be convenient if we could do something like `message.toNSString()` as part of the base library :trollface:

**Review by dgelessus — CHANGES_REQUESTED** [review]
Otherwise this looks good I think.

**Review by dgelessus — APPROVED** [review]

File: Sources/Plasma/NucleusLib/pnUtils/pnUtCrypt.cpp [hunk 3]
```diff
@@ -95,7 +95,11 @@ static void Rc4Codec (
 ) {
     // RC4 uses the same algorithm to both encrypt and decrypt
     uint8_t * temp = (uint8_t *)malloc(bytes);
+
+    IGNORE_WARNINGS_BEGIN("deprecated-declarations")
```
Comment (Hoikas) [inline Sources/Plasma/NucleusLib/pnUtils/pnUtCrypt.cpp:99]: The changes in this file probably belong in another PR

File: Sources/Plasma/PubUtilLib/plMessageBox/hsMessageBox.h [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+      14617 N Newport Hwy
+      Mead, WA   99021
+
+*==LICENSE==*/
+
+#ifndef _hsMessageBox_h_
+#define _hsMessageBox_h_
+
+#include "HeadSpin.h"
+#include <string_theory/string>
```
Comment (Hoikas) [inline Sources/Plasma/PubUtilLib/plMessageBox/hsMessageBox.h:47]: A forward declaration would be sufficient and therefore avoid pulling in heavy headers, such as `type_traits` via `string_theory/string`.

File: Sources/Plasma/PubUtilLib/plResMgr/plResManager.cpp [hunk 5]
```diff
@@ -1366,46 +1367,44 @@ bool plResManager::VerifyPages()
...
         ST::string pageFile = pages[i]->GetPagePath().GetFileName();
-        if (strlen(buf) + pageFile.size() > bufSize - 5)
-        {
-            strcat(buf, "...\n");
-            break;
-        }
-
-        strcat(buf, pageFile.c_str());
-        strcat(buf, "\n");
+        ss << pageFile << ST_LITERAL("\n");
```
Comment (Hoikas) [inline Sources/Plasma/PubUtilLib/plResMgr/plResManager.cpp:1384]:
```suggestion
        ss << pageFile << '\n';
```

File: Sources/Plasma/PubUtilLib/plMessageBox/hsMessageBox.h (hunk 4 above)
Comment (dpogue) [reply Sources/Plasma/PubUtilLib/plMessageBox/hsMessageBox.h:47]: Ahh, now that we got rid of the non-ST::String overloads, I can do that!

File: Sources/Plasma/NucleusLib/pnUtils/pnUtCrypt.cpp (hunk 3 above)
Comment (dpogue) [reply Sources/Plasma/NucleusLib/pnUtils/pnUtCrypt.cpp]: Split to #1476

**Review by Hoikas — APPROVED** [review]
